Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs

```diff
@@ -320,6 +320,28 @@ public void EmbeddedObject_WhenAdditiveExplicit_ShouldThrow()
             Assert.Throws<RealmSchemaValidationException>(() => Realm.GetInstance(conf), $"Embedded object {nameof(EmbeddedLevel3)} is unreachable by any link path from top level objects");
         }
 
+        [Test]
+        public void DeleteRealmWorksIfCalledMultipleTimes()
+        {
+            var config = GetFakeConfig();
+            var openRealm = GetRealm(config);
+            openRealm.Dispose();
+            Assert.That(File.Exists(config.DatabasePath));
+
+            // Race condition:
+            // When trying to delete the Realm it can in some occasions (usually when Sync is involved)
+            // still be in use. To make sure other threads that use the same Realm get scheduled again
+            // and can finish their work before we actually delete the Realm files, we have to wait for
+            // a moment here.
+            // TODO: remove this when https://github.com/realm/realm-core/issues/4762 is resolved.
+            Task.Delay(5).Wait();
```

Comment:
Can we remove this now?

Reply:
I've added the test from https://github.com/realm/realm-dotnet/issues/2455#issuecomment-867641086 into this branch as well (82bae01) which fails. Both tests are similar but the the same.

Testing against `fsa/fix-delete-open-race-v11` (which is https://github.com/realm/realm-core/pull/4768) fixed the problem where we had to add a delay in the `SyncBase` (removed with https://github.com/realm/realm-dotnet/pull/2401/commits/c8b17a4f1182cc308c6ca2b9f3a98c9c7cacbb49) class but it does not fix this specific test.

Only adding the delay fixed it for me. The same with this test case that was already in there. I'm pushing both to CI to verify but it seems like https://github.com/realm/realm-core/pull/4768 does not solve all of those problems yet, as mentioned before. Which makes sense I guess, because https://github.com/realm/realm-core/issues/4762 was the race between deleting the Realm vs. open (which was not related to Sync but a general problem) and on the other side, the test in here is about the Sync part. I guess the misleading part is that I link the same Core issue.

Reply:
After further investigation:
- https://github.com/realm/realm-dotnet/blob/df/use-delete-realm-from-core/Tests/Realm.Tests/Database/InstanceTests.cs#L80 succeeds
- https://github.com/realm/realm-dotnet/blob/df/use-delete-realm-from-core/Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs#L324 fails

They are the same test with one difference:
`var config = RealmConfiguration.DefaultConfiguration;`
vs
`var config = GetFakeConfig();`

Maybe it's actually not Core but our test setup. I need to look into this further.

Reply:
Interestingly I do see the following result:
![image](https://user-images.githubusercontent.com/1942012/123278781-248e8100-d4ff-11eb-8931-c86dc2bf0a8d.png)
100% when run solo, 95% when run together (both tests or the whole class). Sometimes both fail.

Adding to the comment prior to this:
`DeleteRealmWorksIfCalledMultipleTimes` in `SynchronizedInstanceTests` actually always succeeds when run solo.

Reply:
The parameter is only influencing the call to `AddDummyData` btw.

Now it gets interesting:
When I remove that call completely, i.e. no data in the Realm, both tests always fail.

Reply:
How do the tests fail?

Reply:
The only result I've seen is:
```
Realms.Exceptions.RealmInUseException : Cannot delete files of an open Realm. xxxxx.lock is still in use.
```

Reply:
Ah, okay, I see - so we should just retry - there's no way to check what the status of the sync thread is, so we should just retry the deletion 10-20 times and assert that it eventually succeeds.

Reply:
Alright. Is there any value in keeping them then though? Every other test does that anyway in tear down. I would basically just copy paste tear down (and then call tear down on top of that anyway).

Reply:
The tests should not depend on code in teardown to actually test the behavior as refactoring that may result in accidentally removing test coverage. So yes, both tests provide value, we should just explain why we're retrying the deletion in the synchronized case.